### ISSUE TITLE
[Feature] To track the training callback details 

### ISSUE THREAD

**Issue by mahimairaja** [dreamquark-ai/tabnet#508]
## Feature request

**What is the expected behavior?**
To track the training callback details 

**What is motivation or use case for adding/changing the behavior?**
In tensorflow and pytorch we could easily track the performance of the model in a variable history which is a sort of dict like.

**How should this be implemented in your opinion?**
An class similar to History from tf.keras.callback can be written with similar logic

**Are you willing to work on this yourself?**
yes

**Comment by Optimox** [comment]
I am not sure I understand, call backs are already implemented. Moreover, you can access `model.history` to get information about the training

**Comment by mahimairaja** [comment]
No, I am not able to access the callbacks using model.history

**Comment by mahimairaja** [comment]
I just trained the model using the below line:

```python 
history = best_model.fit(xtrain, ytrain, patience=10, \
                max_epochs=50, eval_set=[(xtrain, ytrain)])
```

And I got the below values in a dict when I access with `best_model.history.history.keys()`

```bash
['loss', 'lr', 'val_0_accuracy']
```

But tracking values like - `acc`, `val_acc`, `loss`, `val_loss` could be more feasible as compared to tensorflow

**Comment by Optimox** [comment]
The history will contain all metrics tracked on the different evaluations sets given:
- add (x_valid, y_valid) to your eval_set
- add more metric as the loss you are willing to monitor in your eval_metrics

**Comment by Optimox** [comment]
Please feel free to reopen if needed.
